### PR TITLE
Readme/newcomer infos

### PR DESCRIPTION
**Pull request by isthatcentered**
Hello,

I discovered the library today and found it really amazing. As a newcomer, I found a few informations were missing, so here is my proposal.

This makes the default input/output paths behavior a bit more easier to find and adds typescript to the list of features.

Thank you very much for the amazing work!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,13 +44,41 @@ Microbundle includes two commands - `build` (the default) and `watch`. Neither r
 
 ### `microbundle` / `microbundle build`
 
-By default, microbundle will infer the location of your source entry file
-(the root module in your program) from the `source` field in your `package.json`. It will infer the output directory and filename(s) from the `main` field. For UMD builds, microbundle will use a snake case version of the `name` field in your `package.json` for the export name; you can also specify a name via an `amdName` field or the `name` CLI option.
+Unless overridden via command line, microbundle uses the `source` property in your `package.json` to locate the input file, and the `main` property for the output.
```
Comment (ChrisBrownie55) [inline README.md]:
```suggestion
Unless overridden via the command line, microbundle uses the `source` property in your `package.json` to locate the input file, and the `main` property for the output.
```

File: README.md [hunk 2]
```diff
@@ -16,18 +16,23 @@
...
+
 `npm i -D microbundle`
 
-... then add the scripts to your `package.json`:
+### Set up your `package.json`
 
 ```js
 {
+  "source": "src/foo.js",         // Your source file (same as 1st arg to microbundle)
+  "main": "dist/foo.js",          // Desire output path
```
Comment (ChrisBrownie55) [inline README.md]:
```suggestion
  "main": "dist/foo.js",          // Desired output path
```

File: README.md [hunk 3]
```diff
@@ -39,13 +44,41 @@ Microbundle includes two commands - `build` (the default) and `watch`. Neither r
...
 
-By default, microbundle will infer the location of your source entry file
-(the root module in your program) from the `source` field in your `package.json`. It will infer the output directory and filename(s) from the `main` field. For UMD builds, microbundle will use a snake case version of the `name` field in your `package.json` for the export name; you can also specify a name via an `amdName` field or the `name` CLI option.
+Unless overridden via command line, microbundle uses the `source` property in your `package.json` to locate the input file, and the `main` property for the output.
+
+For UMD builds, microbundle will use a snake case version of the `name` field in your `package.json` as export name. This can be overridden either by providing an `amdName` key in your `package.json` or via the `--name` flag in the cli.
 
 ### `microbundle watch`
 
 Just like `microbundle build`, but watches your source files and rebuilds on any change.
```
Comment (ChrisBrownie55) [inline README.md]:
```suggestion
Acts just like `microbundle build`, but watches your source files and rebuilds on any change.
```

File: README.md [hunk 4]
```diff
@@ -39,13 +44,41 @@ Microbundle includes two commands - `build` (the default) and `watch`. Neither r
...
 
+### Using with typescript
+
+Just point to a `.ts` file as input using either the cli or the `source` key in your `package.json` and you’re done.
```
Comment (ChrisBrownie55) [inline README.md]:
```suggestion
Just point the input to a `.ts` file through either the cli or the `source` key in your `package.json` and you’re done.
```

File: README.md [hunk 5]
```diff
@@ -16,18 +16,23 @@
 - Produces tiny, optimized code for all inputs
 - Supports multiple entry modules _(`cli.js` + `index.js`, etc)_
 - Creates multiple output formats for each entry _(<abbr title="CommonJS (node)">CJS</abbr>, <abbr title="Universal Module Definition">UMD</abbr> & <abbr title="ECMAScript Modules">ESM</abbr>)_
+- 0 configuration Typecript support
```
Comment (ChrisBrownie55) [inline README.md]:
```suggestion
- 0 configuration TypeScript support
```

Comment (ChrisBrownie55) [reply README.md]: You could also say "No" instead of "0" but that's just choice of words.

File: README.md [hunk 6]
```diff
@@ -39,13 +44,41 @@ Microbundle includes two commands - `build` (the default) and `watch`. Neither r
...
-(the root module in your program) from the `source` field in your `package.json`. It will infer the output directory and filename(s) from the `main` field. For UMD builds, microbundle will use a snake case version of the `name` field in your `package.json` for the export name; you can also specify a name via an `amdName` field or the `name` CLI option.
+Unless overridden via command line, microbundle uses the `source` property in your `package.json` to locate the input file, and the `main` property for the output.
+
+For UMD builds, microbundle will use a snake case version of the `name` field in your `package.json` as export name. This can be overridden either by providing an `amdName` key in your `package.json` or via the `--name` flag in the cli.
 
 ### `microbundle watch`
 
 Just like `microbundle build`, but watches your source files and rebuilds on any change.
 
+### Using with typescript
```
Comment (ChrisBrownie55) [inline README.md]:
```suggestion
### Using with TypeScript
```

File: README.md [hunk 7]
```diff
@@ -39,13 +44,41 @@ Microbundle includes two commands - `build` (the default) and `watch`. Neither r
...
+### Specifying builds in `package.json`
+
+You can specify output builds in a `package.json` as follows:
+
+```
+"main": "dist/foo.js",          // CJS bundle
+"umd:main": "dist/foo.umd.js",  // UMD bundle
+"module": "dist/foo.mjs",       // ES Modules bundle
+"source": "src/foo.js",         // custom entry module (same as 1st arg to microbundle)
+"types": "dist/foo.d.ts",       // Typescript typings
```
Comment (ChrisBrownie55) [inline README.md]:
```suggestion
"types": "dist/foo.d.ts",       // TypeScript typings
```

**Comment by ChrisBrownie55** [comment]
If we're modifying the README here could we also address the fact that [`--define` and `--alias` are missing from the README](https://github.com/developit/microbundle/pull/312#issuecomment-473600805)

Basically, it would be nice to update the `All CLI Options` section with the updated output of `microbundle --help`, as well as, provide a short example of how to use `--define` and `--alias` because it isn't currently apparent.

**Comment by isthatcentered** [comment]
@ChrisBrownie55 definitely. 

The cli already provides some nice to the point examples, do you want to use those or do you want to add something a bit more descriptive ?

Here's the cli output
image

File: README.md [hunk 8]
```diff
@@ -16,18 +16,23 @@
...
+
 `npm i -D microbundle`
 
-... then add the scripts to your `package.json`:
+### Set up your `package.json`
 
 ```js
 {
+  "source": "src/foo.js",         // Your source file (same as 1st arg to microbundle)
+  "main": "dist/foo.js",          // Desired output path
```
Comment (developit) [inline README.md]:
```suggestion
  "main": "dist/foo.js",        // output path for CommonJS/Node
  "module": "dist/foo.mjs",     // output path for JS Modules
  "unpkg": "dist/foo.umd.js",   // optional, for unpkg.com
```

File: README.md [hunk 9]
```diff
@@ -16,18 +16,23 @@
...
   "scripts": {
-    "build": "microbundle",
+    "build": "microbundle",       // uses the "source" and "main" key in the file as input and output paths by default
```
Comment (developit) [inline README.md]:
```suggestion
    "build": "microbundle",       // uses "source" and "main" as input and output paths by default
```

**Comment by developit** [comment]
Great work! I left a couple little tweaks but otherwise this looks good to merge!

**Comment by isthatcentered** [comment]
I feel like some informations about the "target" option could be worth adding. 

My guess from looking at the output is that "target web" exposes the module as a global variable which "target node" doesn't. 

Is that right, I did not found an equivalent on Rollup's docs but I might not be looking for the right keyword.

**Pull request merged**